Jenkinsfile

```diff
@@ -1,39 +1,37 @@
-pipeline {
-  agent {
-    label 'vetsgov-general-purpose'
+def notify = { ->
+  if (env.BRANCH_NAME == 'production' ||
+      env.BRANCH_NAME == 'master') {
+    message = "veterans-employment-center ${env.BRANCH_NAME} branch CI failed. |${env.RUN_DISPLAY_URL}".stripMargin()
+    slackSend message: message,
+    color: 'danger',
+    failOnError: true
   }
+}
 
-  stages {
-    stage('Checkout Code') {
-      steps {
-        checkout scm
-      }
-    }
-
-    stage('Install bundle') {
-      steps {
-        sh 'bash --login -c "bundle install -j 4 --without development"'
-      }
-    }
-
-    stage('Ensure database exists') {
-      steps {
-        sh 'bash --login -c "bundle exec rake db:create db:schema:load db:migrate"'
-      }
-    }
-
-    stage('Update bundle-audit database') {
-      steps {
-        sh 'bash --login -c "bundle exec bundle-audit update"'
-      }
-    }
+node('vetsgov-general-purpose') {
+  properties([[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', daysToKeepStr: '60']]]);
 
-    stage('Run tests') {
-      steps {
-        sh 'bash --login -c "bundle exec rake"'
-        sh 'bash --login -c "bundle exec brakeman"'
-        sh 'bash --login -c "bundle exec bundle-audit"'
-      }
+  stage('Setup') {
+    checkout scm
+    sh "docker-compose build --no-cache --force-rm"
+  }
+  stage('Ensure database exists') {
+    sh "docker-compose -p vec up -d"
+    sh "docker-compose -p vec run veteran-employment-center bundle exec rake db:create db:schema:load db:migrate"
+  }
+  // stage('Update bundle-audit database') {
+  //   sh "docker-compose -p vec run veteran-employment-center bundle exec bundle-audit update"
+  // }
+  stage('Run tests') {
+    try {
+      sh "docker-compose -p vec run veteran-employment-center bundle exec rake"
+      sh "docker-compose -p vec run veteran-employment-center bundle exec brakeman"
+      // sh "docker-compose -p vec run veteran-employment-center bundle exec bundle-audit"
+    } catch (err) {
+      notify()
+      throw err
+    } finally {
```

Comment:
I think we want to move this to an `always` block in a `post` stage of the pipeline. I believe that if something were to fail in the earlier stages this would get skipped, but moving it there would ensure it always gets cleaned up.

Reply:
Yes! Overlooked that.